### PR TITLE
Allow to modify dns options of nodes (vms)

### PR DESCRIPTION
**Pull request by mfranczy**
Signed-off-by: Marcin Franczyk <marcin0franczyk@gmail.com>

**How to categorize this PR?**

/area networking
/kind enhancement
/priority normal
/platform kubevirt

**What this PR does / why we need it**:
Allows to modify dns options of nodes

**Which issue(s) this PR fixes**:
Fixes #
https://github.com/gardener/gardener-extension-provider-kubevirt/issues/50

**Special notes for your reviewer**:

**Release note**:

```improvement operator
improvement_operator
```

### CODE REVIEW

**Comment by mfranczy** [comment]
unit tests are failing, currently checking why

**Comment by mfranczy** [comment]
as always I don't have an access to see build logs, I run `make verify-extended` and exit code from it's 0, no issues there. @stoyanr could you tell me what's wrong? Again output from `make generate` is invalid? In that case that would be weird cause I run this time on completely different OS.

**Comment by mfranczy** [comment]
apologies, I do have access to logs, I see that `make verify` fails again, it doesn't on my local machine.

**Comment by mfranczy** [comment]
@stoyanr what are the worker nodes which run concourse? Do you know the operating system and go and helm version?

**Comment by mfranczy** [comment]
I copied controller-registration file from the master branch to avoid delay

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -188,6 +193,8 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"cloudConfig": string(pool.UserData),
 				"kubeconfig":  string(kubeconfig),
 			},
+			"dnsPolicy": workerConfig.DNSPolicy,
+			"dnsConfig": workerConfig.DNSConfig,
```
Comment (stoyanr) [inline pkg/controller/worker/machines.go:197]: I don't see these values used anywhere in the machineclass Helm charts? I don't think adding them here would have any effect without it.

**Review by stoyanr — CHANGES_REQUESTED** [review]
Nice PR, except that something important is missing. Have you tested it?

As a side remark, I don't like that `DNSConfig` is passed as a JSON string in https://github.com/gardener/machine-controller-manager-provider-kubevirt/blob/f3600b1f2296f4df617817f8a67100b96bc4266b/pkg/kubevirt/apis/provider_spec.go#L32, and then unmarshalled from JSON in the code. I would much prefer to use the proper type `corev1.PodDNSConfig` there as well, and to ensure that the YAML in the machineclass conforms to that type so that it's unmarshalled implicitly. Could you please correct this as well?

File: pkg/controller/worker/machines.go (hunk 1 above)
Comment (mfranczy) [reply pkg/controller/worker/machines.go:197]: you're right, good catch.

**Comment by mfranczy** [comment]
> Nice PR, except that something important is missing. Have you tested it?
No, I didn't test it yet, I should add WIP label

**Comment by mfranczy** [comment]
I am going to test this now, will inform about result

**Comment by mfranczy** [comment]
@stoyanr tested and works as expected, I've added info about mcm-provider-kubevirt change we must do https://github.com/gardener/gardener-extension-provider-kubevirt/issues/50#issuecomment-698945770 (i will work on that on Tuesday). For me looks like this PR is good to merge. @moadqassem after merge you could rebase against that.

**Review by stoyanr — APPROVED** [review]

**Pull request merged**